frontend/src/contexts/ErrorContext.tsx

```
@@ -0,0 +1,332 @@
+  backendStatus: 'available',
+  retryCount: 0,
+  lastErrorTime: null,
+  errorHistory: [],
+};
+
+function errorReducer(state: ErrorState, action: ErrorAction): ErrorState {
+  switch (action.type) {
+    case 'SET_GLOBAL_ERROR':
+      const errorId = `error_${Date.now()}_${Math.random().toString(36).substr(2, 9)}`;
+      return {
+        ...state,
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)